pymc3/distributions/continuous.py

```diff
@@ -223,81 +228,64 @@ class Uniform(BoundedContinuous):
         Upper limit.
     """
 
-    def __init__(self, lower=0, upper=1, *args, **kwargs):
-        self.lower = lower = tt.as_tensor_variable(floatX(lower))
-        self.upper = upper = tt.as_tensor_variable(floatX(upper))
-        self.mean = (upper + lower) / 2.0
-        self.median = self.mean
+    @classmethod
+    def dist(cls, lower=0, upper=1, **kwargs):
+        lower = tt.as_tensor_variable(floatX(lower))
+        upper = tt.as_tensor_variable(floatX(upper))
+        # mean = (upper + lower) / 2.0
+        # median = self.mean
 
-        super().__init__(lower=lower, upper=upper, *args, **kwargs)
+        transform = kwargs.get("transform", cls.default_transform)
+        transform = cls.create_transform(transform, lower, upper)
 
-    def random(self, point=None, size=None):
-        """
-        Draw random values from Uniform distribution.
+        rv_var = uniform(lower, upper, **kwargs)
```

Comment:
Will the random methods be part of aesara going forward? This would make it a bit more awkward to add new distributions.

Reply:
No, the `Distribution.random` methods are obsolete under these changes.  Some of the contents of these methods might be used to implement new `RandomVariable` `Op`s, though.